src/Swashbuckle.AspNetCore.Annotations/SwaggerParameterAttribute.cs

```diff
@@ -30,5 +32,37 @@ public bool Required
         }
 
         internal bool? RequiredFlag { get; set; }
+
+        /// <summary>
+        /// Describes how the parameter value will be serialized depending on the type of
+        /// the parameter value. Default values (based on value of in): for query - form;
+        /// for path - simple; for header - simple; for cookie - form.
+        /// </summary>
+        public string Style
+        {
+            get { throw new InvalidOperationException($"Use {nameof(ParameterStyle)} instead"); }
+            set
+            {
+                ParameterStyle = Enum.TryParse(value, ignoreCase: true, out ParameterStyle result) ? result :
+                    throw new InvalidOperationException(
+                        message: $"Style '{value}' not defined in OpenAPI specification");
```

Comment:
I would make this a `NotSupportedException` and tweak the wording. Then if the specification later adds a new value, but we don't support it, then we're not making claims about the specification that are false.